csharp/extractor/Semmle.Extraction.CSharp.Standalone/Razor.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Semmle.BuildAnalyser;
+using Semmle.Util;
+using System.Text;
+
+namespace Semmle.Extraction.CSharp.Standalone
+{
+    internal class Razor
+    {
+        private readonly DotnetVersion sdk;
+        private readonly ProgressMonitor progressMonitor;
+        private readonly DotNet dotNet;
+        private readonly string sourceGeneratorFolder;
+        private readonly string cscPath;
+
+        public Razor(DotnetVersion sdk, DotNet dotNet, ProgressMonitor progressMonitor)
+        {
+            this.sdk = sdk;
+            this.progressMonitor = progressMonitor;
+            this.dotNet = dotNet;
+
+            sourceGeneratorFolder = Path.Combine(this.sdk.FullPath, "Sdks/Microsoft.NET.Sdk.Razor/source-generators");
+            if (!Directory.Exists(sourceGeneratorFolder))
+            {
+                this.progressMonitor.RazorSourceGeneratorMissing(sourceGeneratorFolder);
+                throw new Exception($"Razor source generator folder {sourceGeneratorFolder} does not exist.");
+            }
+
+            cscPath = Path.Combine(this.sdk.FullPath, "Roslyn/bincore/csc.dll");
+            if (!File.Exists(cscPath))
+            {
+                this.progressMonitor.CscMissing(cscPath);
+                throw new Exception($"csc.dll {cscPath} does not exist.");
+            }
+        }
+
+        private static void GenerateAnalyzerConfig(IEnumerable<string> cshtmls, string analyzerConfigPath)
+        {
+            using var sw = new StreamWriter(analyzerConfigPath);
+            sw.WriteLine("is_global = true");
+
+            foreach (var f in cshtmls)
+            {
+                sw.WriteLine($"\n[{f}]");
+                var base64 = Convert.ToBase64String(Encoding.UTF8.GetBytes(f)); // TODO: this should be the relative path of the file.
+                sw.WriteLine($"build_metadata.AdditionalFiles.TargetPath = {base64}");
+            }
+        }
+
+        public IEnumerable<string> GenerateFiles(IEnumerable<string> cshtmls, IEnumerable<string> references, string workingDirectory)
+        {
+            // TODO: the below command might be too long. It should be written to a temp file and passed to csc via @.
```

Comment:
The integration test on Windows is probably failing due to this issue (even though the error message doesn't exactly say that). 

```
Failed to generate source files from cshtml files: An error occurred trying to start process 'dotnet' with working directory 'C:\a\semmle-code\semmle-code\target\codeql-csharp-integration-tests\ql\csharp\ql\integration-tests\all-platforms\cshtml_standalone'. The filename or extension is too long.
```